### PR TITLE
OSD-23318: Allow to modify IDMS in hosted clusters 

### PR DESCRIPTION
**Pull request by samanthajayasinghe**
## What 
As part of zero-egress implementation, hosted clusters IDMS needs to update with private ECR 

- This PR allows mirroring ECR's  for hosted clusters via the feature flag
- backplane service account allows to update IDMS/ICSP/ITMS

## JIRA 
https://issues.redhat.com/browse/OSD-23318

### CODE REVIEW

**Comment by hectorakemp** [comment]
Great work Sam! this all looks good to me :). One question - why did we change the `SyncSetLabelSelector` on the webhook? Was that part of the generated `make` code? 
Happy to `lgtm` if tested :)
/lgtm

File: pkg/webhooks/imagecontentpolicies/imagecontentpolicies.go
Comment (bmeng) [inline pkg/webhooks/imagecontentpolicies/imagecontentpolicies.go:253]: Should we allow the backplane user to do this or a particular service account?

Comment (samanthajayasinghe) [reply pkg/webhooks/imagecontentpolicies/imagecontentpolicies.go:245]: Updated to only allow for SA

**Comment by hectorakemp** [comment]
This generally looks good to me (the CM follows the convention set out in https://github.com/openshift/managed-cluster-validating-webhooks/pull/279). 
/lgtm

**Review by mjlshen — CHANGES_REQUESTED** [review]
/hold

I think we should adjust the approach away from a feature flag if possible. We temporarily added one as a bit of a rush to get a feature out to a subset of the fleet in the past and it would be best if we could avoid having a long-lived implementation. Some potential options:

* Do we even need to deploy this webhook to HCP clusters? I recall that editing this IDMS resource from the hosted cluster itself has no effect, so do we even care if customers can modify it? https://github.com/openshift/managed-cluster-validating-webhooks/blob/b1e1f9bc8e52a6bc55a3eed365bd2cdd8ce12d92/pkg/webhooks/imagecontentpolicies/imagecontentpolicies.go#L172-L174
* Is it feasible to allow the Kubernetes user that `hypershift-operator` is leveraging to always modify these resources instead? The customer should only ever be able to influence this behavior through OCM right and we can shift the guardrails to OCM?

File: pkg/webhooks/imagecontentpolicies/imagecontentpolicies.go
Comment (mjlshen) [inline pkg/webhooks/imagecontentpolicies/imagecontentpolicies.go:40]: Granting ourselves admin access isn't a part of this card - is it needed?

**Comment by samanthajayasinghe** [comment]
Thanks @mjlshen for the review, As discussed, we will disable the webhooks from hosted clusters

**Pull request closed without merging**
